### PR TITLE
Finnish lightning strike state component

### PR DESCRIPTION
**Pull request by shannahutch**
understanding- getInitialState, componentDidMount, setState.  what they and how the work together

### CODE REVIEW

File: react.html [hunk 1]
```diff
@@ -65,10 +65,19 @@
 	  	  	strikes: 0
 	  	  };
   	  },
+  	  timerTick: function() {
+  	  	this.setState({
+  	  		strikes: this.state.strikes + 100
+  	  	});
+  	  },
+  	  componentDidMount: function() {
+  	  	setInterval(this.timerTick, 1000);
```
Comment (mikemintz) [inline react.html]: When this component unmounts, the timer will keep running every second, which may cause errors at the worst, or waste resources at the best. In this situation, you should save the timer id (maybe into state), and define a componentWillUnmount method that calls clearInterval to stop the timer.

You can see an example of this on https://facebook.github.io/react/ under "A Stateful Component". I'll paste it here for posterity:

``` jsx
var Timer = React.createClass({
  getInitialState: function() {
    return {secondsElapsed: 0};
  },
  tick: function() {
    this.setState({secondsElapsed: this.state.secondsElapsed + 1});
  },
  componentDidMount: function() {
    this.interval = setInterval(this.tick, 1000);
  },
  componentWillUnmount: function() {
    clearInterval(this.interval);
  },
  render: function() {
    return (
      <div>Seconds Elapsed: {this.state.secondsElapsed}</div>
    );
  }
});
```

**Comment by mikemintz** [comment]
Indentation still needs to be fixed. Other than that and my comment above about `clearInterval`, LGTM

File: react.html [hunk 2]
```diff
@@ -67,13 +67,13 @@
   	  },
   	  timerTick: function() {
   	  	this.setState({
-  	  		strikes: this.state.strikes + 100
+  	  		strikes: this.state.strikes + 0
```
Comment (mikemintz) [inline react.html]: Why did you change 100 to 0?

**Comment by mikemintz** [comment]
The 2nd commit looks good to me, although the button looks like it doesn't do anything yet.

File: react.html (hunk 2 above)
Comment (shannahutch) [inline react.html]: I wanted it to stop, while I worked on something else,  it was bugging me.  I know it's weird, but it just was.

**Comment by mikemintz** [comment]
The "Set up componentWillUnmount" commit LGTM
